src/main/java/net/minecraftforge/fml/common/langsupport/IPropertyWrapper.java

```diff
@@ -0,0 +1,61 @@
+package net.minecraftforge.fml.common.langsupport;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+
+public interface IPropertyWrapper
+{
+    String getName();
+
+    Class<?> getType();
+
+    void set(Object value, boolean ignoreAccess) throws IllegalAccessException;
+
+    void get() throws IllegalAccessException;
+
+    boolean isAnnotationPresent(Class<? extends Annotation> annotationClass);
+
+    class FieldWrapper implements IPropertyWrapper
+    {
+        private Object instance;
+        private final Field field;
+
+        public FieldWrapper(Field field, Object instance)
+        {
+            this.field = field;
+            this.instance = instance;
+        }
+
+        @Override
+        public String getName()
+        {
+            return field.getName();
+        }
+
+        @Override
+        public Class<?> getType()
+        {
+            return field.getType();
+        }
+
+        @Override
+        public void set(Object value, boolean ignoreAccess) throws IllegalAccessException
+        {
+            if (ignoreAccess)
+                field.setAccessible(true);
```

Comment:
I wasn't entirely sure how to go about this, to be honest. The `IPropertyWrapper`s are specific to instances of the static container, so maybe it's a better idea to have `ignoreAccess` (and `ignoreFinal`, which I forgot to commit >.>) be intrinsic to the wrapper rather than the `get` and `set` calls? Would be required to specify right when getting a container, though.